src/Services/TestGenerationService.cs

```diff
@@ -516,7 +625,104 @@ private void WriteExplicitConstructor(StringBuilder builder, TestGenerationConte
 			}
 		}
 
+		private void WriteTestMethods(StringBuilder builder, TestGenerationContext context)
+		{
+			if (context.MethodDeclarations.Count == 0)
+			{
+				return;
+			}
+
+			string testedObjectReferenceTemplate = this.Settings.GetTemplate(
+				context.TestFramework,
+				context.MockFramework,
+				TemplateType.TestedObjectReference);
+			var testedObjectReference = ReplaceGlobalOrContentTokens(testedObjectReferenceTemplate, context);
+
+			string testedObjectCreationTemplate = this.Settings.GetTemplate(
+				context.TestFramework,
+				context.MockFramework,
+				TemplateType.TestedObjectCreation);
+			var testedObjectCreation = ReplaceGlobalOrContentTokens(testedObjectCreationTemplate, context);
+
+			var testMethodPrefixes = new List<string>();
+
+			int idx = 0;
+
+			foreach (var methodDescriptor in context.MethodDeclarations)
```

Comment:
If there are no public methods it would be better to have a single "generic" test case like with the old code than to have nothing at all. You could have the tested object creation there but no "actions".